src/libsyntax/errors/emitter.rs

```diff
@@ -23,6 +22,23 @@ use std::io;
 use std::rc::Rc;
 use term;
 
+#[derive(Debug)]
+enum Error {
+    Io(io::Error),
+    Term(term::Error),
+}
+
+impl From<io::Error> for Error {
+    fn from(err: io::Error) -> Self {
+        Error::Io(err)
+    }
+}
+
+impl From<term::Error> for Error {
+    fn from(err: term::Error) -> Self {
+        Error::Term(err)
+    }
+}
```

Comment:
This seems pretty unfortunate, shouldn't a `term::Error` be convertible into a `io::Error`?

This same duplication seems to happen in libtest which is what makes me wary